.github/workflows/docker-build-and-test.yml

```diff
@@ -0,0 +1,104 @@
+name: Docker Build and Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      run_tests:
+        description: 'Run tests'
+        required: false
+  push:
+    branches: [ "improvement/docker-testing-workflow-branch" ]  # Isolated branch name for testing to ensure we don't affect main
+
+jobs:
+  build_and_test:
+    if: ${{ github.ref == 'refs/heads/improvement/docker-testing-workflow-branch' || github.event_name == 'workflow_dispatch' }}
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_IMAGE: aisk-unit-tests  # Updated Docker image name
+      DOCKER_TAG: ${{ github.sha }}
+      PROD_MODE: false
+      TEST_MODE: true
+      # Enable debug logging
+      ACTIONS_RUNNER_DEBUG: 'true'
+      ACTIONS_STEP_DEBUG: 'true'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Verify .dockerignore is present
+        run: |
+          if [ ! -f .dockerignore ]; then
+            echo "Error: .dockerignore file is missing."
+            exit 1
+          else
+            echo ".dockerignore file is present."
+          fi
+
+      - name: Create empty .env file
+        run: touch .env
+
+      - name: Build Docker image with BuildKit and verbose logging
+        run: |
+          echo "Building Docker image with BuildKit and verbose logging..."
+          docker build \
```

Comment:
Have we installed the `docker` package? Where?

Reply:
Docker is native to ubuntu see 
[runs on: ](runs-on: ubuntu-latest)

Reply:
Ok thanks I didn't know that.